diff --git a/displayservers/Wayland/gl.c b/displayservers/Wayland/gl.c
index df5e4658..be5b9976 100644
--- a/displayservers/Wayland/gl.c
+++ b/displayservers/Wayland/gl.c
@@ -99,7 +99,11 @@ void waylandEGLSwapBuffers(EGLDisplay display, EGLSurface surface, const struct
       wl_surface_set_buffer_scale(wlWm.surface, 1);
       if (!wlWm.viewport)
         wlWm.viewport = wp_viewporter_get_viewport(wlWm.viewporter, wlWm.surface);
-      wp_viewport_set_source(wlWm.viewport, 0, 0, wlWm.width * wlWm.scale, wlWm.height * wlWm.scale);
+      wp_viewport_set_source(
+          wlWm.viewport,
+          wl_fixed_from_int(-1), wl_fixed_from_int(-1),
+          wl_fixed_from_int(-1), wl_fixed_from_int(-1)
+      );
       wp_viewport_set_destination(wlWm.viewport, wlWm.width, wlWm.height);
     }
     else
